File: 백준/Silver/1817. 짐 챙기는 숌/짐 챙기는 숌.cc
#include <iostream>
#include <vector>
using namespace std;
int main(void) {
    int n,m,wgh=0,cnt=1;
    cin>>n>>m;
    if(n==0) {
        cout<<0;
        return 0;
    }
    vector<int> arr(n);
    for(int i=0;i<n;i++)
        cin>>arr[i];
    for(int i=n-1;i>=0;i--) {
        wgh+=arr[i];
        if(wgh>m) {
            wgh=arr[i];
            cnt+=1;
        }
    }
    cout<<cnt;
    return 0;
}

File: 백준/Silver/1817. 짐 챙기는 숌/짐 챙기는 숌.py
import sys
input=lambda:sys.stdin.readline().rstrip()
n,m=map(int,input().split())
arr=[*map(int,input().split())]
wgh,cnt=0,1
if n>0:
    for i in range(n-1,-1,-1):
        wgh+=arr[i]
        if wgh>m:
            wgh=arr[i]
            cnt+=1
    print(cnt)
else:
    print(0)